lib/spack/spack/util/binary_resource.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+"""
+Interface for bootstrapping needed resources for running of Spack;
+unlike bootstrap.py (the typical pathway) this does not refer to
+utilities with Specs, but rather by an ID that refers to the needed binary.
+
+This is currently only needed on Windows.
+"""
+
+import os
+import pathlib
+import shutil
+
+from llnl.util import tty
+
+import spack.paths
+import spack.util.path as sup
+
+BINARY_RESOURCE_SUBDIR = "binary-resources"
+
+
+def binary_resource_root() -> pathlib.Path:
+    """Returns the root of the Windows resources required for bootstrapping"""
+    return pathlib.Path(spack.paths.user_cache_path) / BINARY_RESOURCE_SUBDIR
+
+
+def win_insert_resource_into_environment(name):
+    resource_root = binary_resource_root() / name
+    env = spack.util.environment.EnvironmentModifications()
+    env.append_path("PATH", str(resource_root / "bin"))
+    env.apply_modifications()
+
+
+class BinaryResource:
+    """Represents a resource required by Spack to run, fetched as part
+    of the bootstrapping operation
+
+    More or less identical to package based resources but exposes additional logic related
+    to bootstrapping/ resource visibility.
+
+    Composes a name and a fetch strategy
+    """
+
+    def __init__(self, name, conf):
+        """
+        Args:
+            name (str): Name of resource to be fetched
+            conf (dict): Dictionary representing resource endpoint layout
+        """
+        self._name = name
+        self.resource_subdir = BINARY_RESOURCE_SUBDIR
+        fetcher = spack.fetch_strategy.URLFetchStrategy(
+            url=sup.substitute_path_variables(conf["endpoint"]), checksum=conf["sha256"]
+        )
+        stage = spack.stage.Stage(fetcher)
+        resource = spack.resource.Resource(
+            name, fetcher, destination=self.resource_subdir, placement=None
+        )
+        self.stage = spack.stage.ResourceStage(fetcher, stage, resource, keep=False)
```

Comment:
(question) I think we spoke about using `Stage` directly vs. constructing a `ResourceStage`, did that not work out?